applications/DAOsign.md

```diff
@@ -0,0 +1,186 @@
+# DAOsign
+
+- **Team Name:** DAOsign 
+- **Payment Address:** 13s3c37rwrZfa8WQcZPdqE1zi1hGAguf2TWfWMr8f5aLBrg
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 3
+
+## Project Overview :page_facing_up:
+
+
+### Overview
+
+DAOsign is a decentralized platform for agreements and policies with publicly verifiable and authorized cryptographic proofs and Governance Automation, designed for DAOs. 
+Traditional methods of signing agreements, such as paper-based systems or legacy e-signatures, are prone to fraud and lack transparency, and offer no solution for verification of the signer's identity or signer’s authority to execute the agreement. Instead these systems rely on costly legal proceedings for enforce agreements. 
+
+DAOsign provides publicly verifiable cryptographic proofs; secure, transparent and automated processing of complex Agreements and Policies. 
+
+Furthermore, DAOsign provides robust Governance and Policy Builder platform to provide governance, workflow automation and controls for organization policies and procedures.
+### Project Details
+A project goal is to store Agreements Proofs on Polkadot chain using Ink! smart contracts. 
+
+
+Deliveries include DAOsign platform with an ability to:
+- create agreements including generating using ChatGPT
+- add and remove signers and observers of different levels of anonymity
+- allow signing the documents with further storing of signature proofs on Astar testnet blockchain.
+
+
+**Core DAOsign components:**
+
+<img width="902" alt="Screenshot 2023-05-27 at 16 51 43" src="https://github.com/Burmenska/Grants-Program/assets/3983591/e6e3e78d-edf4-4d50-a298-e01210d75963">
+
+
+**Technical design:**
+
+Proofs are implemented in the EIP-712 standard and adjusted to ink! ecosystem. Proofs digned using user’s EVM private key. 
+We are supporting this standard in Ink! smart contract to be able to verify the signature in ink! smart contract. 
+Also, we are implementing an adjusted EIP-2771 standard to allow Agreement Signers gas-free delegation to publish proof on-chain.
+EIP-712 standard will be used for proofs, adaptation means that user data is converted to a binary bit; we use its hash (serialization vs deserialization)
+And EIP-2771 standard - to securely send the proof and receive it.
+So that EVM standards will be adapted to ink! ecosystem requirements. 
+Example (interfaces) of the smart contracts can be found here https://github.com/DAOsign/daosign-smart-contracts  (Solidity version).
+
+
+Detailed storyboard can be found here: https://consideritdone.storiesonboard.com/m/PUCLDvQC4kKwoRJ8kbm3MA .
+
+Architectural diagram:
+
+<img width="805" alt="Screenshot 2023-05-28 at 23 30 53" src="https://github.com/Burmenska/Grants-Program/assets/3983591/54c8708d-e180-46f3-9a78-b7121de133d3">
```

Comment:
What exactly does the _Agreement Proof_ component represent?

Reply:
Also, what is the _Forwarder_? Is it the off-chain component that handles and relays the meta txs?

Reply:
> Also, what is the _Forwarder_? Is it the off-chain component that handles and relays the meta txs?

Correct, it is off-chain component to relay txs. We included this component into the scope.

> What exactly does the Agreement Proof component represent?
In DAOsign, Agreement Proof entity represent 3 proofs:
* Proof of signature
* Proof of authority
* Proof of agreement

I am attaching a document where we described technical design for proofs

[Proofs Technical Design - DAOSign.pdf](https://github.com/w3f/Grants-Program/files/12183424/Proofs.Technical.Design.-.DAOSign.pdf)

Reply:
Thanks. Could you incorporate this information into the proposal as well?

Reply:
Done

Reply:
@ramilexe I haven't found the integration of this:

> Correct, it is off-chain component to relay txs. We included this component into the scope.

Also, the _2. | Gas Relayer | Off-chain component for transaction relaying_ deliverable seems to describe the same concept as the forwarder. Are they identical? Could you clarify?

Reply:
Also, regarding the _Agreement Proof_: in your [docs](https://github.com/w3f/Grants-Program/files/12183424/Proofs.Technical.Design.-.DAOSign.pdf) you define is as _"object that contains Agreement File Proof CID along with Agreement Sign Proof CIDs of all Signers"_. I'm a bit confused to why you would use an object or data structure in place of an actor in your sequence diagram. Could you clarify which actor would create these proofs? I assume it'd be the same actor that returns the _agreement id_ and the boolean _agreement result_ to the _creator_.

Reply:
@ramilexe Just something to keep in mind: I'm not sure if there's currently an easy way to use metamask/secp256k1 accounts on ink!

Reply:
> Also, the _2. | Gas Relayer | Off-chain component for transaction relaying_ deliverable seems to describe the same concept as the forwarder. Are they identical? Could you clarify?

Updated seq diagram and added description. The reference is [EIP-2771](https://eips.ethereum.org/EIPS/eip-2771)

Reply:
> Also, regarding the Agreement Proof: in your [docs](https://github.com/w3f/Grants-Program/files/12183424/Proofs.Technical.Design.-.DAOSign.pdf) you define is as "object that contains Agreement File Proof CID along with Agreement Sign Proof CIDs of all Signers". I'm a bit confused to why you would use an object or data structure in place of an actor in your sequence diagram. Could you clarify which actor would create these proofs? I assume it'd be the same actor that returns the agreement id and the boolean agreement result to the creator.

`AgreementProof` in my diagram is a smart contract, added some description there